tests/devices/qubit/test_simulate.py

```diff
@@ -1178,3 +1184,228 @@ def test_qinfo_tf(self):
 
         grad5 = grad_tape.jacobian(results[5], phi)
         assert qml.math.allclose(grad5, expected_grads[5])
+
+
+class TestMidMeasurements:
+    """Tests for simulating scripts with mid-circuit measurements using the ``simulate_tree_mcm``."""
+
+    @pytest.mark.parametrize("val", [0, 1])
+    def test_basic_mid_meas_circuit(self, val):
+        """Test execution with a basic circuit with mid-circuit measurements."""
+        qs = qml.tape.QuantumScript(
+            [qml.Hadamard(0), qml.CNOT([0, 1]), qml.measurements.MidMeasureMP(0, postselect=val)],
+            [qml.expval(qml.X(0)), qml.expval(qml.Z(0))],
+        )
+        result = simulate_tree_mcm(qs)
+        assert result == (0, (-1.0) ** val)
+
+    def test_basic_mid_meas_circuit_with_reset(self):
+        """Test execution with a basic circuit with mid-circuit measurements."""
+        qs = qml.tape.QuantumScript(
+            [
+                qml.Hadamard(0),
+                qml.Hadamard(1),
+                qml.CNOT([0, 1]),
+                (m0 := qml.measure(0, reset=True)).measurements[0],
+                qml.Hadamard(0),
+                qml.CNOT([1, 0]),
+            ],  # equivalent to a circuit that gives equiprobable basis states
+            [qml.probs(op=m0), qml.probs(op=qml.Z(0)), qml.probs(op=qml.Z(1))],
+        )
+        result = simulate_tree_mcm(qs)
+        assert qml.math.allclose(result, qml.math.array([0.5, 0.5]))
+
+    # pylint: disable=too-many-arguments
+    @pytest.mark.parametrize("shots", [None, 5500])
+    @pytest.mark.parametrize("postselect", [None, 0])
+    @pytest.mark.parametrize("reset", [False, True])
+    @pytest.mark.parametrize("measure_f", [qml.counts, qml.expval, qml.probs, qml.sample, qml.var])
+    @pytest.mark.parametrize(
+        "meas_obj",
+        [qml.Y(0), [1], [1, 0], "mcm", "composite_mcm", "mcm_list"],
+    )
+    def test_simple_dynamic_circuit(self, shots, measure_f, postselect, reset, meas_obj):
```

Comment:
Why is this included in this PR?

Reply:
I added it here because I am removing a similar one from `default_qubit_native_mcm` in this [PR](https://github.com/PennyLaneAI/pennylane/pull/6246)